### ISSUE TITLE
Savestate saving broken on AARCH64 libretro core (Error during serialize)

### ISSUE THREAD

**Issue by joyrider3774** [tiberiusbrown/Ardens#118]
Hi,

This [commit](https://github.com/tiberiusbrown/Ardens/commit/6ff57bdb1dc0e60919f9e92d84562f5dff450e5a) about "Time travel debugging" seems to have broken save state saving on at least aarch64 build libretro core. I builded the latest version of the core for muos on rg35xxsp and it kept giving errors about "error during serialize" while trying to save state. If build the [commit](https://github.com/tiberiusbrown/Ardens/commit/b85f0cd32d9b0746eedd8b60b279ac716abb09c6) before that it still works fine.

The errors i get are these :
```
[libretro INFO] Calculated serialize size: 16865532
[libretro ERROR] Error during serialize
[ERROR] [State]: Failed to save state to "/run/muos/storage/save/state/Ardens/1944.state1".
[INFO] [State]: Loading state "/run/muos/storage/save/state/Ardens/1944.state1", 0 bytes.
[ERROR] [State]: Failed to load state from "/run/muos/storage/save/state/Ardens/1944.state1".
[INFO] [State]: Loading state "/run/muos/storage/save/state/Ardens/1944.state1", 0 bytes.
[ERROR] [State]: Failed to load state from "/run/muos/storage/save/state/Ardens/1944.state1".
[libretro ERROR] Error during serialize
[ERROR] [State]: Failed to save state to "/run/muos/storage/save/state/Ardens/1944.state2".
[libretro ERROR] Error during serialize
[ERROR] [State]: Failed to save state to "/run/muos/storage/save/state/Ardens/1944.state3".
[libretro ERROR] Error during serialize
[ERROR] [State]: Failed to save state to "/run/muos/storage/save/state/Ardens/1944.state4".
[libretro ERROR] Error during serialize
[ERROR] [State]: Failed to save state to "/run/muos/storage/save/state/Ardens/1944.state5".
```

Cmake commandline used to build the core
```
 cmake .. -DARDENS_LIBRETRO=1 -DCMAKE_TOOLCHAIN_FILE=/opt/aarch64-buildroot-linux-gnu_sdk-buildroot/share/buildroot/toolchainfile.cmake
```

the retroarch version i'm using is
```
[/]# retroarch --version
RetroArch - Frontend for libretro
Version: 1.19.1 Jun  5 2024
Compiler: GCC (13.2.0) 64-bit
```

**Comment by tiberiusbrown** [comment]
Fix included in v0.24.2. Thanks for the report!

**Comment by joyrider3774** [comment]
hi thanks, for the fix, i can confirm it now works again
